### PR TITLE
Add .gitattributes file to prevent test's and spec's downloading, add correct getting channelId

### PR DESCRIPTION
**Pull request by kuai6**
Hi there. 
I just add .gitattributes  file to prevent test's, spec's and tutorials downloading when composer install. Because  https://www.reddit.com/r/PHP/comments/2jzp6k/i_dont_need_your_tests_in_my_production .

Also, i add correct getting channelId and able to reuse already opened channels. 
I change `\Bunny\AbstractClient::channel` method for backward capability.

### CODE REVIEW

**Comment by jakubkulhan** [comment]
Hello, thank for the PR. I'll be happy to merge changes regarding `.gitattributes`. However, I don't see channel reuse getting merged. If you need to reuse the channel, just use the instance returned the first time :)

**Comment by kuai6** [comment]
Hello. It's realy need becouse if i need open more than one channel (in different places), i need make my own channel pool or just use already established connection.

**Comment by jakubkulhan** [comment]
Yes, channel pool would be solution. However, it must be separated from the client class. Both are separate responsibilities.

**Comment by kuai6** [comment]
Hi. I made code  changes. Please check

**Comment by kuai6** [comment]
Hi. Please check latest changes )

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/benchmark/ export-ignore
+/spec/ export-ignore
+/test/ export-ignore
+/tutorial  export-ignore
+/.travis.yml export-ignore
+/phpunit.xml export-ignore
+/.gitattributes export-ignore
+/.gitignore export-ignore
```
Comment (jakubkulhan) [inline .gitattributes]: missing eol

File: src/Bunny/AbstractClient.php [hunk 2]
```diff
@@ -66,6 +66,12 @@
     /** @var int */
     protected $frameMax = 0xFFFF;
 
+    /** @var int  */
+    protected $nextChannelId = 1;
+
+    /** @var int  */
+    protected $channelMax = 255;
```
Comment (jakubkulhan) [inline src/Bunny/AbstractClient.php]: I haven't found this limit anywhere in the spec - if there is no limit, channel max should be `0xFFFF` as channel ID is 16bit number.

File: src/Bunny/AbstractClient.php [hunk 3]
```diff
@@ -364,12 +370,15 @@ public function getState()
      *
      * Channel gets first available channel id.
      *
+     * @param bool $forceOpen Open new channel without check existent ready channels
```
Comment (jakubkulhan) [inline src/Bunny/AbstractClient.php]: forgotten doc-comment

File: src/Bunny/AbstractClient.php [hunk 4]
```diff
@@ -447,4 +456,39 @@ protected function getFrameMax()
...
+            ++$channelId
+        ) {
+            if (!isset($this->channels[$channelId])) {
+                $this->nextChannelId = $channelId + 1;
+
+                return $channelId;
+            }
+        }
+
+        throw new ClientException('No available channels');
```
Comment (jakubkulhan) [inline src/Bunny/AbstractClient.php]: `'` -> `"` (use quotes instead of apostrophe for strings)

File: src/Bunny/Channel.php [hunk 5]
```diff
@@ -756,5 +756,4 @@ protected function onBodyComplete()
             throw new \LogicException("Either return or deliver frame has to be handled here.");
         }
     }
-
```
Comment (jakubkulhan) [inline src/Bunny/Channel.php]: please leave the blank line

**Comment by jakubkulhan** [comment]
Also I've updated `.travis.yml` in master so builds won't fail. Please rebase onto master.

**Comment by kuai6** [comment]
Done)

**Pull request merged**